test/e2e/util.go

```diff
@@ -63,6 +65,59 @@ type TestContextType struct {
 
 var testContext TestContextType
 
+type QueueItem struct {
```

Comment:
Can you please move this to a separate file fifo_queue.go? This file is already quite long.